src/openrct2/actions/BannerSetStyleAction.cpp

```diff
@@ -163,5 +163,8 @@ GameActions::Result BannerSetStyleAction::Execute() const
     intent.PutExtra(INTENT_EXTRA_BANNER_INDEX, _bannerIndex);
     ContextBroadcastIntent(&intent);
 
+    ScrollingTextInvalidate();
```

Comment:
The only reason to why the banner colors work now is because of this line, TTF does not process the color and adding color to the cache doesn't make any sense. The problem is the cache used in ScrollingTextSetup.